base_layer/core/src/transactions/transaction_components/wallet_output.rs

```diff
@@ -239,6 +249,65 @@ impl WalletOutput {
         ))
     }
 
+    /// It creates a transaction input given a partial script signature. The public keys
+    /// `partial_script_public_key` and `partial_total_nonce` exclude caller's private keys
+    pub async fn as_transaction_input_with_partial_signature<KM: TransactionKeyManagerInterface>(
+        &self,
+        factory: &CommitmentFactory,
+        partial_script_public_key: PublicKey,
+        partial_total_nonce: PublicKey,
+        key_manager: &KM,
+    ) -> Result<TransactionInput, TransactionError> {
+        let value = self.value.into();
+        let commitment = key_manager.get_commitment(&self.spending_key_id, &value).await?;
+
+        let version = TransactionInputVersion::get_current_version();
+        let script_nonce_a = PrivateKey::random(&mut OsRng);
+        let script_nonce_b = PrivateKey::random(&mut OsRng);
+        let ephemeral_commitment = factory.commit(&script_nonce_b, &script_nonce_a);
+        // TODO: Is this correct? It seems `PublicKey::default()` should be repalced with some non-default value
+        let ephemeral_pubkey = PublicKey::default() + &partial_total_nonce;
```

Comment:
I dont think it is, default = 0, so we should be able to leave that out and just assign `partial_total_nonce`